hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/HeartbeatManager.java

```diff
@@ -16,92 +16,56 @@
 
 /**
- * HeartbeatManager manager used by connection manager.
+ * Periodically on each `heartbeatInterval` pings active connections.
+ * Connections not receiving any packages for `heartbeatTimeout` duration
+ * is closed with  {@link TargetDisconnectedException}.
  */
-public class HeartbeatManager implements Runnable {
-
-    private final TcpClientConnectionManager clientConnectionManager;
-    private final HazelcastClientInstanceImpl client;
-    private final ILogger logger;
-    private final long heartbeatInterval;
-    private final long heartbeatTimeout;
-    private final ClientICMPManager clientICMPManager;
-
-    HeartbeatManager(TcpClientConnectionManager clientConnectionManager, HazelcastClientInstanceImpl client) {
-        this.clientConnectionManager = clientConnectionManager;
-        this.client = client;
-        HazelcastProperties properties = client.getProperties();
-        this.heartbeatTimeout = properties.getPositiveMillisOrDefault(HEARTBEAT_TIMEOUT);
-        this.heartbeatInterval = properties.getPositiveMillisOrDefault(HEARTBEAT_INTERVAL);
-        this.logger = client.getLoggingService().getLogger(HeartbeatManager.class);
-        ClientIcmpPingConfig icmpPingConfig = client.getClientConfig().getNetworkConfig().getClientIcmpPingConfig();
-        this.clientICMPManager = new ClientICMPManager(icmpPingConfig,
-                (ClientExecutionServiceImpl) client.getTaskScheduler(),
-                client.getLoggingService(), clientConnectionManager, this);
-    }
-
-    public void start() {
-        client.getTaskScheduler()
-                .scheduleWithRepetition(this, heartbeatInterval, heartbeatInterval, MILLISECONDS);
-        clientICMPManager.start();
-    }
-
-    long getHeartbeatTimeout() {
-        return heartbeatTimeout;
-    }
+public final class HeartbeatManager {
 
-    @Override
-    public void run() {
-        if (!clientConnectionManager.isAlive()) {
-            return;
-        }
+    private HeartbeatManager() {
 
-        long now = Clock.currentTimeMillis();
-        for (final ClientConnection connection : clientConnectionManager.getActiveConnections()) {
-            checkConnection(now, connection);
-        }
     }
 
-    private void checkConnection(long now, final ClientConnection connection) {
-        if (!connection.isAlive()) {
-            return;
-        }
+    public static void start(HazelcastClientInstanceImpl client,
+                             TaskScheduler taskScheduler,
+                             ILogger logger,
+                             long heartbeatIntervalMillis,
+                             long heartbeatTimeoutMillis,
+                             ActiveConnections activeConnections) {
+        taskScheduler.scheduleWithRepetition(() -> {
+            for (Connection connection : activeConnections.get()) {
+                if (!connection.isAlive()) {
+                    return;
+                }
+                long now = Clock.currentTimeMillis();
```

Comment:
Minor: I don't have some arguments to back that, but I personally liked the version that we get the current time once better